scripts/src/main/java/com/mathewsachin/fategrandautomata/scripts/modules/Support.kt

```diff
@@ -393,4 +409,36 @@ class Support(fgAutomataApi: IFGAutomataApi) : IFGAutomataApi by fgAutomataApi {
         val mlbSimilarity = prefs.support.mlbSimilarity
         return limitBreakRegion.exists(limitBreakPattern, Similarity = mlbSimilarity)
     }
+
+    private fun skillLevels(bounds: Region, skillLevels: List<Boolean>): Boolean {
+        val y = bounds.Y + 325
+        val x = bounds.X + 1600
+
+        val skillLoc = listOf(
+            Location(x + 2 + 25, y),
+            Location(x + 158 + 25, y),
+            Location(x + 312 + 25, y)
+        )
+
+        val result = skillLoc.withIndex().map {
+            if (!skillLevels[it.index])
+                true
+            else {
+                val skillRegion = Region(it.value, Size(35, 45))
+                skillRegion.exists(images.skillTen, Similarity = 0.68)
+            }
+        }
+
+        val skillString = result.withIndex().joinToString("/") {
+            val maxReq = skillLevels[it.index]
+            when {
+                !maxReq -> "x"
+                it.value -> "10"
+                else -> "f"
+            }
+        }
+        println(skillString)
```

Comment:
Shouldn't this be logd?

Reply:
we don't have logd here. We're in an non-Android project 😢

Reply:
I didn't want to add an interface and pass it right now.

Reply:
I thought about it back then, maybe we can get the same kind of logs with https://github.com/tony19/logback-android, which relies on the Slf4j API. Slf4j can also be used in Java modules.

Reply:
slf4j has an official library for Android: http://www.slf4j.org/android/
We don't need to use logback then.

Reply:
last commit on slf4j-android was 7 yrs ago 😢

Reply:
But still, nothing speaks against using it, if it works.

Reply:
I'm a bit confused. Maybe they're using some different repo now.
Last release was in Dec'19
https://mvnrepository.com/artifact/org.slf4j/slf4j-android

Reply:
Beware of this: https://stackoverflow.com/a/35012588/5971497

Reply:
yeah, I'd seen that issue. It's for Verbose. We're logging at debug level, so it should be fine.

Reply:
In the linked Slf4j JIRA ticket, there's a reference to this library, which is also getting updates and actually logs debug messages: https://gitlab.com/mvysny/slf4j-handroid

Reply:
> It's for Verbose

Nope, I've read that everything below INFO is not logged.

Reply:
yeah, you're right. I tried running it and didn't log. I'll try `slf4j-handroid` then.

Reply:
done